### PR TITLE
cryptominisat 5.7.1

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

remake of #53731
deprecate support for high-sierra machines

### CODE REVIEW

File: Formula/cryptominisat.rb [hunk 1]
```diff
@@ -12,11 +11,16 @@ class Cryptominisat < Formula
   end
 
   depends_on "cmake" => :build
+  # fix `'stdio.h' file not found` on macosx-10.13
+  depends_on :xcode => ["11.0", :build]
```
Comment (SMillerDev) [inline Formula/cryptominisat.rb:15]: Does it actually require XCode? Or just sierra or higher?

Comment (chenrui333) [reply Formula/cryptominisat.rb:15]: yeah, I can do mojave, now I think that might be a better way to solve this issue.

File: Formula/cryptominisat.rb [hunk 2]
```diff
@@ -12,11 +11,16 @@ class Cryptominisat < Formula
   end
 
   depends_on "cmake" => :build
+  # fix `'stdio.h' file not found` on macosx-10.13
+  depends_on :macos => :mojave
```
Comment (SMillerDev) [inline Formula/cryptominisat.rb:15]: Is there a way to do this only on macOS?

**Comment by miccal** [comment]
@chenrui333 version 5.8.0 is now available:
```
url "https://github.com/msoos/cryptominisat/archive/5.8.0.tar.gz"
sha256 "50153025c8503ef32f32fff847ee24871bb0fc1f0b13e17fe01aa762923f6d94"
```

**Comment by chenrui333** [comment]
Closed in favor of #58435

**Pull request closed without merging**